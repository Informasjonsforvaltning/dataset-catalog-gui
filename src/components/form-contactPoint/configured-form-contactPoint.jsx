import { reduxForm } from 'redux-form';

import FormContactPoint from './form-contactPoint.component';
import validate from './form-contactPoint-validations';
import { asyncValidateDatasetInvokePatch } from '../../pages/dataset-registration-page/formsLib/asyncValidateDatasetInvokePatch';

const config = {
  form: 'contactPoint',
  validate,
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormTitle = reduxForm(config)(FormContactPoint);
