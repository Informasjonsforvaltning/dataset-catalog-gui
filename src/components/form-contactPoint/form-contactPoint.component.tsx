import React, { memo, FC } from 'react';
import { compose } from 'redux';
import { Field } from 'redux-form';

import {
  withTranslations,
  Props as TranslationsProps
} from '../../providers/translations';

import Helptext from '../helptext/helptext.component';
import InputField from '../fields/field-input/field-input.component';
import InputFieldReadonly from '../fields/field-input-readonly/field-input-readonly.component';
import LinkReadonlyField from '../fields/field-link-readonly/field-link-readonly.component';

interface ExternalProps {
  isReadOnly: boolean;
}

interface Props extends ExternalProps, TranslationsProps {}

const FormContactPoint: FC<Props> = ({ isReadOnly, translationsService }) => (
  <form>
    <div className='form-group'>
      <Helptext
        title={translationsService.translate(
          'schema.contactPoint.helptext.organizationalUnit'
        )}
        term='ContactPoint_organizational-unit'
        recommended
      />
      <Field
        name='contactPoint[0].organizationUnit'
        component={isReadOnly ? InputFieldReadonly : InputField}
        label={translationsService.translate(
          'schema.contactPoint.organizationUnitLabel'
        )}
        isOnlyOneSelectedLanguage
      />
    </div>
    <div className='form-group'>
      <Helptext
        title={translationsService.translate(
          'schema.contactPoint.helptext.hasURL'
        )}
        term='ContactPoint_hasURL'
      />
      <Field
        name='contactPoint[0].hasURL'
        component={isReadOnly ? LinkReadonlyField : InputField}
        label={translationsService.translate('schema.contactPoint.hasURLLabel')}
        isOnlyOneSelectedLanguage
      />
    </div>
    <div className='form-group'>
      <Helptext
        title={translationsService.translate(
          'schema.contactPoint.helptext.hasEmail'
        )}
        term='ContactPoint_hasEmail'
      />
      <Field
        name='contactPoint[0].email'
        component={isReadOnly ? InputFieldReadonly : InputField}
        label={translationsService.translate('schema.contactPoint.emailLabel')}
        isOnlyOneSelectedLanguage
      />
    </div>
    <div className='form-group'>
      <Helptext
        title={translationsService.translate(
          'schema.contactPoint.helptext.hasTelephone'
        )}
        term='ContactPoint_hasTelephone'
      />
      <div className='w-50'>
        <Field
          name='contactPoint[0].hasTelephone'
          component={isReadOnly ? InputFieldReadonly : InputField}
          label={translationsService.translate(
            'schema.contactPoint.hasTelephoneLabel'
          )}
          isOnlyOneSelectedLanguage
        />
      </div>
    </div>
  </form>
);

export default compose<FC<ExternalProps>>(
  memo,
  withTranslations
)(FormContactPoint);
