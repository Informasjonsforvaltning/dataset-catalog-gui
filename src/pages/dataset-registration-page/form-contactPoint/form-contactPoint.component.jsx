import React from 'react';
import { Field } from 'redux-form';
import PropTypes from 'prop-types';
import localization from '../../../services/localization';
import { Helptext } from '../../../components/helptext/helptext.component';
import InputField from '../../../components/fields/field-input/field-input.component';
import InputFieldReadonly from '../../../components/fields/field-input-readonly/field-input-readonly.component';
import LinkReadonlyField from '../../../components/fields/field-link-readonly/field-link-readonly.component';

export const FormContactPoint = ({ isReadOnly }) => (
  <form>
    <div className="form-group">
      <Helptext
        title={localization.schema.contactPoint.helptext.organizationalUnit}
        term="ContactPoint_organizational-unit"
      />
      <Field
        name="contactPoint[0].organizationUnit"
        component={isReadOnly ? InputFieldReadonly : InputField}
        label={localization.schema.contactPoint.organizationUnitLabel}
        isOnlyOneSelectedLanguage
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.contactPoint.helptext.hasURL}
        term="ContactPoint_hasURL"
      />
      <Field
        name="contactPoint[0].hasURL"
        component={isReadOnly ? LinkReadonlyField : InputField}
        label={localization.schema.contactPoint.hasURLLabel}
        isOnlyOneSelectedLanguage
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.contactPoint.helptext.hasEmail}
        term="ContactPoint_hasEmail"
      />
      <Field
        name="contactPoint[0].email"
        component={isReadOnly ? InputFieldReadonly : InputField}
        label={localization.schema.contactPoint.emailLabel}
        isOnlyOneSelectedLanguage
      />
    </div>
    <div className="form-group">
      <Helptext
        title={localization.schema.contactPoint.helptext.hasTelephone}
        term="ContactPoint_hasTelephone"
      />
      <div className="w-50">
        <Field
          name="contactPoint[0].hasTelephone"
          component={isReadOnly ? InputFieldReadonly : InputField}
          label={localization.schema.contactPoint.hasTelephoneLabel}
          isOnlyOneSelectedLanguage
        />
      </div>
    </div>
  </form>
);
FormContactPoint.propTypes = {
  isReadOnly: PropTypes.bool
};
FormContactPoint.defaultProps = {
  isReadOnly: false
};
