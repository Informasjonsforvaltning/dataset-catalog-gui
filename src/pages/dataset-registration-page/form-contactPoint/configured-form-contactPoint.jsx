import { reduxForm } from 'redux-form';

import { FormContactPoint } from './form-contactPoint.component';
import validate from './form-contactPoint-validations';
import { asyncValidateDatasetInvokePatch } from '../formsLib/asyncValidateDatasetInvokePatch';

const config = {
  form: 'contactPoint',
  validate,
  asyncValidate: asyncValidateDatasetInvokePatch
};

export const ConfiguredFormTitle = reduxForm(config)(FormContactPoint);
