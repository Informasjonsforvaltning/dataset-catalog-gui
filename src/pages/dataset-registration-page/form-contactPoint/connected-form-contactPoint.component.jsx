import { getFormSyncErrors } from 'redux-form';
import { connect } from 'react-redux';
import _ from 'lodash';

import { ConfiguredFormTitle } from './configured-form-contactPoint';
import { contactPointType } from '../../../schemaTypes';

const mapStateToProps = (state, ownProps) => {
  const { datasetItem } = ownProps;
  return {
    initialValues: {
      contactPoint:
        _.get(datasetItem, 'contactPoint', []).length > 0
          ? _.get(datasetItem, 'contactPoint')
          : [contactPointType]
    },
    syncErrors: getFormSyncErrors('contactPoint')(state)
  };
};

export const ConnectedFormContactPoint = connect(mapStateToProps)(
  ConfiguredFormTitle
);
