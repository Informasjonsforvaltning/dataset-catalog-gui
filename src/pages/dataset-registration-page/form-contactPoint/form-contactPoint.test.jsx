import React from 'react';
import { shallow } from 'enzyme';
import { FormContactPoint } from './form-contactPoint.component';

let defaultProps;
let wrapper;

beforeEach(() => {
  defaultProps = {};
  wrapper = shallow(<FormContactPoint {...defaultProps} />);
});

test('should render FormContactPoint correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
