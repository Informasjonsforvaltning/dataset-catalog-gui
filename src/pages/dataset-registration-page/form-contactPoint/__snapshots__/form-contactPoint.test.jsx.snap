// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormContactPoint correctly 1`] = `
<form>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="ContactPoint_organizational-unit"
      title="Kontaktpunkt"
    />
    <Field
      component={[Function]}
      isOnlyOneSelectedLanguage={true}
      label="Kontaktpunkt"
      name="contactPoint[0].organizationUnit"
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="ContactPoint_hasURL"
      title="Kontaktskjema"
    />
    <Field
      component={[Function]}
      isOnlyOneSelectedLanguage={true}
      label="Kontaktskjema"
      name="contactPoint[0].hasURL"
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="ContactPoint_hasEmail"
      title="E-post"
    />
    <Field
      component={[Function]}
      isOnlyOneSelectedLanguage={true}
      label="E-post"
      name="contactPoint[0].email"
    />
  </div>
  <div
    className="form-group"
  >
    <withStateHandlers(Component)
      term="ContactPoint_hasTelephone"
      title="Telefon"
    />
    <div
      className="w-50"
    >
      <Field
        component={[Function]}
        isOnlyOneSelectedLanguage={true}
        label="Telefon"
        name="contactPoint[0].hasTelephone"
      />
    </div>
  </div>
</form>
`;
